plugins/besecure.rb

```diff
@@ -0,0 +1,310 @@
+        return ''
+      end
+      
+      if @debug
+        print_status(res)
+      end
+
+      body = ''
+      begin
+        body = JSON.parse(res.body)
+      rescue JSON::ParserError
+        print_error("#{@hostname} - Unable to parse the response")
+        return ''
+      end
+  
+      if body['error']
+        print_error("#{@hostname} - An error occured:")
+        print_error(body)
+        return ''
+      end
+
+      data = body['data']
+      data.each do |item|
+        tbl << [ item['ID'], item['Name'], item['PrettyRange']]
+      end
+
+      # print_good(body)
+  
+      print_good("beSECURE list of reports")
+      print_line
+      print_line tbl.to_s
+      print_line
+    end
+
+    def cmd_besecure_report_download(*args)
+      if args?(args, 4)
+        req = Net::HTTP::Post.new('/json.cgi', initheader={'Host'=>@hostname})
+        format_file = args[1]
+        req.set_form_data({'apikey' => @apikey, 'primary' => 'vulnerabilities', 'secondary' => 'report', 'action' => 'getreport', 'network' => args[0], 'format' => format_file})
+
+        if @debug
+          print_status(req.body)
+        end
+
+        http = Net::HTTP::new(@hostname, 443)
+        http.use_ssl = true
+        res = http.start {|http| http.request(req)}
+
+        unless res
+          print_error("#{@hostname} - Connection timed out")
+          return ''
+        end
+        
+        if @debug
+          print_status(res)
+        end
+
+        body = ''
+        begin
+          body = JSON.parse(res.body)
+        rescue JSON::ParserError
+          print_error("#{@hostname} - Unable to parse the response")
+          return ''
+        end
+    
+        if body['error']
+          print_error("#{@hostname} - An error occured:")
+          print_error(body)
+          return ''
+        end
+
+        # if @debug
+        #   print_status(body)
+        # end
+
+        decompressed = ''
+        if format_file != 'json'
+          compressed_base64 = body["compresseddata"]
+          compressed = Base64.decode64(compressed_base64)
+          decompressed = Zlib::Inflate.inflate(compressed)
+        else
+          decompressed = body
+        end
+
+        if @debug
+          print_status(decompressed)
+        end
+
+        ::FileUtils.mkdir_p(args[2])
+        name = ::File.join(args[2], args[3])
+        print_status("Saving report to #{name}")
+        output = ::File.new(name, "w")
+        output.puts(decompressed)
+        output.close
+      else
+        print_status("Usage: besecure_report_download <network_id> <format_name> <path> <report_name>")
+      end
+    end
+
+    def cmd_besecure_report_import(*args)
+      if args?(args, 2)
+        tempfile = Tempfile.new('results')
+
+        cmd_besecure_report_download(args[0], 'nbe', File.dirname(tempfile) + "/", File.basename(tempfile) )
+
+        print_status("Importing report to database.")
+        framework.db.import_file({:filename => tempfile})
```

Comment:
For the `FC3A290A` report on the test account provided, the nbe file is empty. It might be worthwhile adding a guard clause here to detect this scenario, as framework doesn't allow empty import files:

```
[-] Error while running command besecure_report_import: The data provided to the import function was empty
```

Reply:
I will add a test to make sure its not empty

Reply:
Ah, after some research it turns out that the `db_import` command in msfconsole essentially has try/catch logic surrounding the import call:

https://github.com/rapid7/metasploit-framework/blob/6034f48e8fa35f943051c0be85109f0973254fc6/lib/msf/ui/console/command_dispatcher/db.rb#L1503-L1514

Which removes the stack trace pasted above, and handles it slightly more gracefully:

```
msf5 > db_import file_with_a_single_newline.nbe
[-] Failed to import /Users/user/Documents/code/metasploit-framework/file_with_a_single_newline.nbe: The data provided to the import function was empty
```

Reply:
I added a check to make sure the content is non-empty prior to trying to import